synapse/util/caches/dictionary_cache.py

```diff
@@ -253,7 +248,11 @@ def invalidate(self, key: KT) -> None:
         # raced with the INSERT don't update the cache (SYN-369)
         self.sequence += 1
 
-        # Del-multi accepts truncated tuples.
+        # We want to drop all information about the dict for the given key, so
+        # we use `del_multi` to delete it all in one go.
+        #
+        # We ignore the type error here `del_mutli` accepts a truncated key
+        # (when the key type is a tuple).
```

Comment:
```suggestion
        # We ignore the type error here: `del_multi` accepts a truncated key
        # (when the key type is a tuple).
```

It feels like this is a problem with the type annotation on `del_multi`, but maybe let's gloss over that for now.

Reply:
Agreed. I ended up in a bit of a rabbithole when trying to fix the type annotations, but maybe we could do something tactical there.